src/accel/AlongStepFactory.cc

```diff
@@ -61,18 +59,16 @@ auto UniformAlongStepFactory::operator()(AlongStepFactoryInput const& input) con
         }
 
         // Convert field units from tesla to native celeritas units
-        for (real_type& v : field)
+        for (real_type& v : get_field_().field)
```

Comment:
We should call `get_field_` only once in this function: what if it returns a copy?

Also, if we're really accepting the field *parameters* as the result, I don't think we want to scale those in place. The `celeritas/field` stuff should be unchanged once it's created, and in native celeritas units. I guess the purpose of this is to expose the field driver options through `accel`? You asked me about units at some point; was that about  this use case?

Maybe we should have a separate `DriverOptionsFunction` as a "power user" option for specifying the (celeritas native) options when needed. I definitely think we shouldn't be changing the units of `UniformFieldParams` on the fly.

Reply:
Right, the goal was to be able to specify the field driver options based on user input. I asked about units partly because I wasn't sure at what point it made the most sense to do the conversion -- but you're right that it's not here.